.github/workflows/rust.yml

```diff
@@ -27,7 +25,8 @@ jobs:
     steps:
     - uses: actions/checkout@v3
     - run: rustup update
-    - run: cargo fmt --all --check
+    - run: cargo fmt --check
+    - run: cargo clippy -- -Dwarnings -Dclippy::all -Dclippy::pedantic
```

Comment:
Clippy actually builds all of the code, it's like `cargo check` or `cargo build` plus additional linters, so I don't think it should live in the fmt step. The fmt step is intended to be a fast fail on formatting issues, where as clippy will build the entire application which will take a while.

Also by moving clippy out of the build-and-test step, it's not actually running a build other than the builds that the test command runs.